auth/src/main/java/com/firebase/ui/auth/ui/account_link/SaveCredentialsActivity.java

```diff
@@ -60,11 +60,19 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.save_credentials_layout);
+
+        // If SmartLock is not enabled, finish immediately
+        if (!getIntent().getBooleanExtra(ExtraConstants.EXTRA_SMARTLOCK_ENABLED, true)) {
```

Comment:
I think it would be cleaner to just not start this activity if smartlock is not enabled, even though there are a few entry points that would need updating.  Thoughts?